src/dataframe/dataframe.jl

```diff
@@ -121,6 +121,8 @@ function DataFrame(pairs::Pair{Symbol,<:Any}...; makeunique::Bool=false)::DataFr
     DataFrame(columns, Index(colnames, makeunique=makeunique))
 end
 
+DataFrame(dict::Dict{Symbol,<:Any}) = DataFrame((x for x ∈ dict)...)
```

Comment:
Oh, this was really interesting I forgot to mention it.  0.7 throws a deprecation warning without this explicit constructor because the "automatic conversion" constructors are deprecated, but for the life of me I have no idea what the `Dict` constructor actually wound up calling before.  My first instinct was to do away with the `Pair` constructor and have it feed into a `Dict` constructor, but I had to do it the other way around because of `makeunique`.

Reply:
OK. Then `Base.convert(::Type{DataFrame}, d::AbstractDict)` was called automatically on 0.6. On 0.7, the recommendation added in https://github.com/JuliaLang/julia/pull/23273 is to define constructors, and have `convert` methods call constructors explicitly (so the inverse of what we do now, mostly).

Reply:
(Note that splatting should generally be avoided when the number of elements can be large, as is the case here: it's much more efficient to pass the collection and loop over it inside the function.)